server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -359,6 +544,14 @@ protected String contentType() {
     @Override
     protected void doMerge(Mapper mergeWith, boolean updateAllTypes) {
         super.doMerge(mergeWith, updateAllTypes);
+        TextFieldMapper mw = (TextFieldMapper) mergeWith;
+        if (this.prefixFieldMapper != null && mw.prefixFieldMapper != null) {
+            this.prefixFieldMapper = (PrefixFieldMapper) this.prefixFieldMapper.merge(mw.prefixFieldMapper, updateAllTypes);
+        }
+        else if (this.prefixFieldMapper != null || mw.prefixFieldMapper != null) {
+            throw new IllegalArgumentException("mapper [" + name() + "] has different index_prefix settings, current ["
+                + this.prefixFieldMapper + "], merged [" + mw.prefixFieldMapper + "]");
```

Comment:
does prefixFieldMapper have a nice toString?

Reply:
Have added one, plus a line to the test